### PR TITLE
Add Updates to GraphQL v2

### PR DESCRIPTION
**Pull request by sbinlondon**
Necessary PR for https://github.com/opencollective/opencollective-frontend/pull/5319

* Adds Updates object to v2
* Allows you to query Updates on an Account

### CODE REVIEW

File: server/graphql/v2/interface/Account.js [hunk 1]
```diff
@@ -269,6 +270,14 @@ const accountFieldsDefinition = () => ({
       return collective;
     },
   },
+  updates: {
+    type: UpdateCollection,
```
Comment (Betree) [inline server/graphql/v2/interface/Account.js:274]:
```suggestion
    type: new GraphQLNonNull(UpdateCollection),
```

File: server/graphql/v2/interface/Account.js [hunk 2]
```diff
@@ -562,6 +571,28 @@ export const AccountFields = {
       return get(collective.data, 'categories', []);
     },
   },
+  updates: {
+    type: UpdateCollection,
+    args: {
+      limit: { type: GraphQLInt },
+      offset: { type: GraphQLInt },
```
Comment (Betree) [inline server/graphql/v2/interface/Account.js:578]: Can be replaced by:

```suggestion
      ...CollectionArgs,
```

Benefits:
- Add default value on `limit` => we prefer that people don't fetch all expenses at once
- Add `description`s (better docs)

File: server/graphql/v2/object/Update.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          }
+          return req.remoteUser && req.remoteUser.canSeeUpdates(update.CollectiveId);
+        },
+      },
+      isPrivate: { type: new GraphQLNonNull(GraphQLBoolean) },
+      title: { type: new GraphQLNonNull(GraphQLString) },
+      createdAt: { type: new GraphQLNonNull(DateString) },
+      updatedAt: { type: new GraphQLNonNull(DateString) },
+      publishedAt: { type: DateString },
+      image: { type: GraphQLString },
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:61]: This field is never set in the DB, looks like a feature that was never fully implemented. Let's not expose it

```suggestion
```

File: server/graphql/v2/object/Update.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      views: { type: GraphQLInt },
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:62]: This one is not defined on the model

```suggestion
```

File: server/graphql/v2/object/Update.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      markdown: {
+        type: GraphQLString,
+        resolve(update, _, req) {
+          if (update.isPrivate && !(req.remoteUser && req.remoteUser.canSeeUpdates(update.CollectiveId))) {
+            return null;
+          }
+
+          return stripHTML(update.markdown || '');
+        },
+      },
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:94]: This one is deprecated, let's not use it

```suggestion
```

File: server/graphql/v2/object/Update.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      tags: { type: new GraphQLList(GraphQLString) },
+      fromCollective: {
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:96]:
```suggestion
      fromAccount: {
```

File: server/graphql/v2/object/Update.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        type: Account,
+        resolve(update, args, req) {
+          return req.loaders.Collective.byId.load(update.FromCollectiveId);
+        },
+      },
+      collective: {
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:102]:
```suggestion
      account: {
```

File: server/graphql/v2/object/Update.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        resolve(update, args, req) {
+          return req.loaders.Collective.byId.load(update.CollectiveId);
+        },
+      },
+      tier: {
+        type: Tier,
+        resolve(update, args, req) {
+          return req.loaders.Tier.byId.load(update.TierId);
+        },
+      },
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:113]: This field is never set in the DB, looks like a feature that was never fully implemented. Let's not expose it

```suggestion
```

File: server/graphql/v2/object/Update.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from 'graphql';
+import { GraphQLDateTime } from 'graphql-iso-date';
+
+import { stripHTML } from '../../../lib/sanitize-html';
+import { DateString } from '../../v1/types';
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:13]: We should rather use the `GraphQLDateTime` standard, `DateString` is a type from v1

File: server/graphql/v2/object/Update.js [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+import { getIdEncodeResolver, IDENTIFIER_TYPES } from '../identifiers';
+import { Account } from '../interface/Account';
+
+import { Tier } from './Tier';
+
+const UpdateAudienceTypeEnum = new GraphQLEnumType({
```
Comment (Betree) [inline server/graphql/v2/object/Update.js:19]: We should move that to the `enum` folder, it's meant to be used by the `createMutation` too when we'll have it on v2

**Review by Betree — COMMENTED** [review]
This will also resolve #4786 :rocket:

File: server/graphql/v2/interface/Account.js (hunk 2 above)
Comment (sbinlondon) [reply server/graphql/v2/interface/Account.js:578]: 1. Do you have an example of this in use?
2. Why do I keep seeing the word 'expenses' used with Updates? How are they related? I don't see the connection.

File: server/graphql/v2/object/Update.js (hunk 9 above)
Comment (sbinlondon) [reply server/graphql/v2/object/Update.js:13]: Ok, it formats it a bit different from the v1 but that's not a problem. I'll just need to double check that it still plays nice with the frontend.

File: server/graphql/v2/interface/Account.js (hunk 2 above)
Comment (Betree) [reply server/graphql/v2/interface/Account.js:578]: https://github.com/opencollective/opencollective-api/blob/5cd771d0436ab06fe5ef34eaca7bd7ef2ab517ec/server/graphql/v2/query/TransactionsQuery.ts#L14

**Review by Betree — APPROVED** [review]
Looks good!

**Pull request merged**